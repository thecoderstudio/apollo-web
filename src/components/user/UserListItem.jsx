import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import OutlinedButton from '../buttons/OutlinedButton';

const propTypes = {
  agent: PropTypes.object.isRequired
};

const Container = styled.div`
  min-height: 60px;
  padding: 0px 16px;
  margin: 0px;
  display: grid;
  grid-template-columns: 3fr 1fr;
  align-items: center;
`;

const Username = styled.p`
  margin: 0px;
`;

const Tag = styled.p`
  float: right;
  width: 80px;
  padding: 4px;
  font-weight: bold;
  text-align: center;
  border-radius: 6px;
  background-color: ${props => props.theme.accent};
`;

const DeleteButton = styled(OutlinedButton)`
  grid-column: logout;
  float: right;
  width: 100px;
  padding 4px;
  border: 1px solid ${props => props.theme.error};
`;

function UserListItem(props) {
  let role;
  if (props.user.role !== null) {
    role = <Tag>{props.user.role.name}</Tag>;
  }

  return (
    <Container>
      <Username>{props.user.username}</Username>
      {role}
      <DeleteButton>Delete</DeleteButton>
    </Container>
  );
}

UserListItem.propTypes = propTypes;

export default UserListItem;
