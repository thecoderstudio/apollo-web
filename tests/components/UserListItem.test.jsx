import React from 'react';
import renderer from 'react-test-renderer';
import UserListItem from '../../src/components/user/UserListItem';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

function getComponent(user, store) {
  return renderer.create(
    <Provider store={store}>
      <UserListItem user={user} />
    </Provider>
  );
}

describe("user list item", () => {
  const store = mockStore({
    currentUser: {
      role: {
        id: 'id',
        name: 'admin'
      }
    }
  });
  it("render correctly without role", () => {
    const user = {
      username: 'test',
      role: null
    };
    const tree = getComponent(user, store).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render correctly with role", () => {
    const user = {
      username: 'test',
      role: {
        name: 'admin'
      }
    };
    const tree = getComponent(user, store).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
