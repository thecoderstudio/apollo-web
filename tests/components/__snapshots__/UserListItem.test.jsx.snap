// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user list item render correctly with role 1`] = `
.c0 {
  min-height: 60px;
  padding: 0px 16px;
  margin: 0px;
  display: grid;
  grid-template-columns: 3fr 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin: 0px;
}

.c2 {
  float: right;
  width: 80px;
  padding: 4px;
  font-weight: bold;
  text-align: center;
  border-radius: 6px;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    test
  </p>
  <p
    className="c2"
  >
    admin
  </p>
</div>
`;

exports[`user list item render correctly without role 1`] = `
.c0 {
  min-height: 60px;
  padding: 0px 16px;
  margin: 0px;
  display: grid;
  grid-template-columns: 3fr 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin: 0px;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    test
  </p>
</div>
`;
